mmdeploy/apis/visualize.py

```diff
@@ -71,21 +71,29 @@ def visualize_model(model_cfg: Union[str, mmengine.Config],
     with torch.no_grad():
         result = model.test_step(model_inputs)[0]
 
+    visualize = True
     try:
         # check headless
         import tkinter
         tkinter.Tk()
+    except Exception as e:
+        from mmdeploy.utils import get_root_logger
+        logger = get_root_logger()
+        logger.warn(
+            f'render and display result skipped for headless device, exception {e}'  # noqa: E501
+        )
+        visualize = False
+
+    if visualize is True:
+        data_samples = None
```

Comment:
no need to pass `data_samples` since it is not used by https://github.com/open-mmlab/mmdeploy/blob/cc6cf211797850ddeddf90626273b751b2631a90/mmdeploy/codebase/mmdet3d/deploy/voxel_detection.py#L186

Reply:
done.